packages/angular_devkit/build_angular/src/angular-cli-files/models/webpack-configs/common.ts

```diff
@@ -294,9 +294,10 @@ export function getCommonConfig(wco: WebpackConfigOptions) {
     },
     module: {
       rules: [
-        { test: /\.html$/, loader: 'raw-loader' },
+        { test: /\.(html|component.svg)$/, loader: 'raw-loader' },
```

Comment:
Special naming rules should be avoided.

By changing the replace resources transformer [here](https://github.com/angular/angular-cli/blob/master/packages/ngtools/webpack/src/transformers/replace_resources.ts#L142) to inline the raw loader usage, the need for the naming convention can be avoided.  The raw-loader rule could actually be removed completely with such a change.

This is untested, but something similar to the following should work:
```
return `!raw-loader!${/^\.?\.\//.test(node.text) ? '' : './'}${node.text}`;
```

Reply:
I tested your idea, it works. However, the same code is also used for loading html templates and styles. Loaders for css preprocessors would not get invoked any more. It would also make it impossible to customize loaders for .html templates.

If we only use raw-loader here when the template is an .svg file, then we would not break existing code. Something like:
```
if (/\.svg$/.test(node.text)) {
    return `!raw-loader!${/^\.?\.\//.test(node.text) ? '' : './'}${node.text}`;
} else {
    return `${/^\.?\.\//.test(node.text) ? '' : './'}${node.text}`;
}
```
What do you think? If everyone agrees, I'll update the PR with that change and replace the unit tests I added for the previous solution. I would add tests in replace_resources_spec.ts. For the change in compiler_host.ts, I'm not sure yet what the best place to test this would be, as there is no compiler_host_spec.ts.

Reply:
The suggested line above should only be applied to `templateUrl` augmentations.  `styleUrls` behavior should remain unchanged.  The plugin has an option named `directTemplateLoading` which is enabled by the CLI and would be the guard for the inline 'raw-loader` usage.

Reply:
`getResourceUrl` is called by `getRequireExpresion` which is also used when processing the `styleUrls` property in `visitComponentMetadata`. That's why I suggested that `if`. 

I could also add an argument to `getRequireExpression` and `getResourceUrl` that determines if raw-loader should be used.

It could look something like this:
```
// in visitComponentMetadata:
case 'templateUrl':
            return ts.updatePropertyAssignment(node, ts.createIdentifier('template'), createRequireExpression(node.initializer, '!raw-loader!'));

// createRequireExpression
function createRequireExpression(node: ts.Expression, loader = ''): ts.Expression {
  const url = getResourceUrl(node, loader);
// ...

export function getResourceUrl(node: ts.Expression, loader = ''): string | null {
  // only analyze strings
  if (!ts.isStringLiteral(node) && !ts.isNoSubstitutionTemplateLiteral(node)) {
    return null;
  }

  return `${loader}${/^\.?\.\//.test(node.text) ? '' : './'}${node.text}`;
}
```

Reply:
After discussing with the team, this solution appears viable.  Can you make the updates and then we can proceed with another round of reviews?

Reply:
Done. I hope I did not miss any of the comments.